MySQLdb/_mysql.c

```diff
@@ -1284,9 +1284,15 @@ _mysql__fetch_row(
         if (!self->use)
             row = mysql_fetch_row(self->result);
         else {
+            // see: https://docs.python.org/3/library/gc.html#gc.get_referrers
+            // This function can get a reference to the tuple r, and if that
+            // code is preempted while holding a ref to r, the _PyTuple_Resize
+            // will raise a SystemError because the ref count is 2.
+            PyObject_GC_UnTrack(*r);
```

Comment:
Move these code to _mysql_ResultObject_fetch_row.

Reply:
its pretty safe everywhere else we do it in the dangerous area where we allow threads.

Reply:
It is safe untrack soon after creatin, and track again after _PyTuple_Resize().

Reply:
For example, `convert_row` may execute Python code.
Execute python code means it allows threads switching,  Calling gc module, etc.

Py_BEGIN_ALLOW_THREADS is not the only danger API.  There are many danger APIs
which can execute any Python code.  e.g. Py_DECREF, PyObject_RichCompareBool,
PyList_Append, etc.

Reply:
ping?